src/Service/PlanningService.php

```diff
@@ -60,15 +62,14 @@ public function getPlanningData(): PlanningData
      *
      * @return ArrayCollection<string, Weeks> The ArrayCollection of Weeks objects representing the weeks in the planning
      */
-    private function buildPlanningWeeks(PlanningData $planning): ArrayCollection
+    private function buildPlanningWeeks(PlanningData $planning, int $year): ArrayCollection
     {
         $weeks = $planning->weeks;
 
         $now = new \DateTime();
-        $currentYear = (int) $now->format('Y');
-        $currentWeek = (int) $now->format('W');
+        $currentWeek = ($year !== (int) $now->format('Y')) ? null : (int) $now->format('W');
```

Comment:
I don't understand this line. Maybe something along the lines of:

```suggestion
            $selectedDate = (new \DateTime())->setDate($year, (int) $now->format('m'), (int) $now->format('d'));
            $currentWeek = (int) $selectedDate->format('W');
```

Reply:
I only want currentWeek (which is used to highlight the current week in the table) if the year is the current year. If not, i want currentWeek to be null, so that no week will be highlighted.